mailpoet/mailpoet
# WC Tracker - campaign revenue tracking [MAILPOET-5014]

## Description
We want to find out how MailPoet campaigns help to generate revenue in Woo stores.
In order to do so, we want to use WC Tracker for tracking campaign revenue stats data we already store in the `mailpoet_statistics_woocommerce_purchases` table.

### Tracked data
Here is an example of the data that will be added to the tracker data:
```JSON
{
    "campaigns_count": 476,
    "currency": "CZK",
    "campaign_191_revenue": 1200.5,
    "campaign_191_orders_count": 12,
    "campaign_191_type": "standard",
    "campaign_192_revenue": 600,
    "campaign_192_orders_count": 6,
    "campaign_192_type": "standard",
    "campaign_194_revenue": 500,
    "campaign_194_orders_count": 5,
    "campaign_194_type": "welcome",
    "campaign_195_revenue": 800,
    "campaign_195_orders_count": 8,
    "campaign_195_type": "automation",
    "campaign_196_revenue": 600,
    "campaign_196_orders_count": 6,
    "campaign_196_type": "automatic",
    "campaign_197_revenue": 600.5,
    "campaign_197_orders_count": 6,
    "campaign_197_type": "notification",
    "campaign_201_revenue": 700,
    "campaign_201_orders_count": 7,
    "campaign_201_type": "notification"
}
```

`"campaign_XXX_revenue": 1200.0` - Sum of revenue generated by the campaign
`"campaign_XXX_type": "standard"` - Campaign type
`"campaign_XXX_orders_count": 12` - Number of orders that were created after users visited the site via the campaign

Regarding the format, please see [this comment](https://github.com/mailpoet/mailpoet/pull/4665#issuecomment-1401610401).

#### Notes
1. In the DB table, we save revenue for all clicks a subscriber did a couple of days before making an order. In the tracking data, I take into account only the first click so that every order is included only once.
2. `campaign_revenues` contains only campaigns that generated some orders.
3. We save revenue for post notifications (e.g., weekly post notification) emails separately for each sending. I grouped those data under post-notification parent so that all post notifications emails are under a single campaign.
4. In the future, I would like to extend the campaign data with some additional details (e.g., type of an automatic email, type of dynamic segment used, post notifications frequency, and sent count)

## Code review notes

_N/A_

## QA notes

There are no changes in UI.
In case you want to check what is tracked (you need to have WP CLI):
1) Setup page with Woo and MailPoet
2) Create a subscriber and add to a list
3) Send a newsletter to the list
4) Click on a link from the received email
5) Make an order when being signed in using the subscribers email (or use the email in the order)
6) Go to a command line and use: `wp wc tracker snapshot --format=yaml`
7) You should see the data from above being printed among other tracker data

## Linked PRs

_N/A_

## Linked tickets

[MAILPOET-5014]

## After-merge notes

We need to update Woo usage tracking page https://href.li/?https://woocommerce.com/usage-tracking/


[MAILPOET-5014]: https://mailpoet.atlassian.net/browse/MAILPOET-5014?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
